File: 1582-special-positions-in-a-binary-matrix/special-positions-in-a-binary-matrix.c
int numSpecial(int** mat, int matSize, int* matColSize) {
    int cnt = 0;
    
    for (int i = 0; i < matSize; i++) {
        for (int j = 0; j < *matColSize; j++) {
            if (mat[i][j] == 1) {
                bool rowHasOther = false;
                bool colHasOther = false;
                
                for (int k = 0; k < *matColSize; k++) {
                    if (k != j && mat[i][k] == 1) {
                        rowHasOther = true;
                        break;
                    }
                }
                
                for (int k = 0; k < matSize; k++) {
                    if (k != i && mat[k][j] == 1) {
                        colHasOther = true;
                        break;
                    }
                }
                
                if (!rowHasOther && !colHasOther) {
                    cnt++;
                } else {
                    break;
                }
            }
        }
    }
    
    return cnt;
}

File: 1582-special-positions-in-a-binary-matrix/special-positions-in-a-binary-matrix.py
class Solution(object):
    def numSpecial(self, mat):
        rows, cols = len(mat), len(mat[0])
        cnt = 0

        for i in range(rows):
            for j in range(cols):
                if mat[i][j] == 1:
                    if any(mat[i][k] == 1 for k in range(cols) if k != j):
                        break
                    else:
                        if any(mat[k][j] == 1 for k in range(rows) if k != i):
                            break
                        else:
                            cnt += 1
        
        return cnt
